### PR TITLE
Simple status messages

### PR DESCRIPTION
**Pull request by andorp**
* Machinery to render status messages for the user
 * One message on saving quorums
 * `pageFrame` uses the `Frame` type as a simplification.

### CODE REVIEW

**Comment by andorp** [comment]
@fisx: Please translate the message.

File: src/Frontend/Core.hs [hunk 1]
```diff
@@ -416,23 +416,24 @@ form formHandler = getH :<|> postH
 
 -- | Wrap anything that has 'ToHtml' and wrap it in an HTML body with complete page.
 data Frame body
-    = Frame { _frameUser :: User, _frameBody :: body }
+    = Frame { _frameUser :: User, _frameBody :: body, _frameMessages :: [StatusMessage] }
```
Comment (fisx) [inline src/Frontend/Core.hs]: After logout, we are on a public frame, and yet it makes sense to display a message in the frame like "thank you for playing".  But we can change that once we need it.

File: src/Frontend/Core.hs [hunk 2]
```diff
@@ -416,23 +416,24 @@ form formHandler = getH :<|> postH
...
 
 makeFrame :: (ActionPersist m, ActionUserHandler m, MonadError ActionExcept m, Page p)
           => m p -> m (Frame p)
 makeFrame mp = do
   isli <- isLoggedIn
+  msgs <- flushMessages
   let isPrivate = isPrivatePage mp -- Here 'm' is used as the 'proxy'.
   if | not isli && isPrivate -> redirect . absoluteUriPath $ relPath P.Login
-     | isli     || isPrivate -> Frame <$> currentUser <*> mp
+     | isli     || isPrivate -> Frame <$> currentUser <*> mp <*> pure msgs
```
Comment (fisx) [inline src/Frontend/Core.hs]: ...  or perhaps we should do something: this means that logout flushes the message queue without showing it.  Ok, but not ideal.

File: src/Frontend/Page/Admin.hs [hunk 3]
```diff
@@ -288,7 +288,9 @@ adminQuorum :: ActionM m => FormPageHandler m PageAdminSettingsQuorum
 adminQuorum =
     FormPageHandler
         (PageAdminSettingsQuorum <$> query (view dbQuorums))
-        (update . SaveQuorums)
+        (\qs -> do
+            update $ SaveQuorums qs
+            addMessage "Quorums are saved.")
```
Comment (fisx) [inline src/Frontend/Page/Admin.hs]: or:

```haskell
         (update . SaveQuorums >=> const (addMessage "Quorums are saved."))
```

(What would you name something that behaves to `>>` as `>=>` behaves to `>>=`?)

We can keep your version if you prefer.

File: src/Frontend/Core.hs [hunk 4]
```diff
@@ -412,7 +412,7 @@ form formHandler = getH :<|> postH
     processor2 page result = absoluteUriPath . relPath . redirectOf page <$> processor result
 
 
--- * frames
+-- * frame creation
```
Comment (fisx) [inline src/Frontend/Core.hs]: This commit is great, I am glad I recovered the diff even though you did a good job hiding it.  (Spoiiler: `pageFrame` now takes a frame and is much easier to call from the `ToHtml` instance and elsewhere.)

Comment (fisx) [inline src/Frontend/Core.hs]: Please disregard last comment.  It made no sense.

**Comment by fisx** [comment]
Nice work, thanks!

Please merge if you are happy with my changes, or pass back to me with more comments.

File: src/Frontend/Page/Admin.hs (hunk 3 above)
Comment (andorp) [inline src/Frontend/Page/Admin.hs]: I would keep my version, as we will create the `Action.Idiom` module soon.

**Comment by andorp** [comment]
LGTM, first we should merge the #402

**Pull request merged**